src/context/context.js

```diff
@@ -0,0 +1,40 @@
+import React, {Component} from 'react'
+import {db} from '../firebase.js'
+
+const UserContext = React.createContext()
+
+class UserProvider extends Component {
+  state = {
+    shoppingList: []
+  }
+
+  componentDidMount(){
+    db.collection('shoppingList').get()
+      .then(querySnapshot => {
+        const data = querySnapshot.docs.map(doc => doc.data());
+        console.log(data)
+        this.setState({shoppingList: data})
+      })
+  }
```

Comment:
If we refactor this to use a functional component, you'd probably be looking at using a `useEffect` hook (because it is what has essentially replaced `componentDidMount` and `componentDidUpdate`) here to grab that data! [Here is more info should a refactor be in the future!](https://reactjs.org/docs/hooks-effect.html)